school_evaluation/models/school_evaluation.py

```diff
@@ -164,21 +163,33 @@ class SchoolEvaluationTemplate(models.Model):
     type = fields.Selection([('faculty', 'Faculty'), ('student', 'Student')],
                             'User Type', required=True, default='faculty',
                             help="Type of Evaluation")
-    rating_line = fields.One2many('rating.rating', 'rating_id', 'Rating')
+    # rating_line = fields.One2many('rating.rating', 'rating_id', 'Rating')
+    rating_line_ids = fields.Many2many('rating.rating', 'rating_table')
 
 
 class RatingRating(models.Model):
-    _name = 'rating.rating'
-    _description = "Rating details"
-    _rec_name = 'point'
-    _order = "point desc"
+    _inherit = 'rating.rating'
+
+    @api.model
+    def create(self, vals):
+        res_model = self.env['school.evaluation.template'].browse(
+            vals.get('res_id'))
+        vals.update({'res_model_id': res_model.id})
```

Comment:
mention the reason plz

Reply:
to set Document Model (res_model_id)  which is mandatory for the rating(which is inherited from rating.rating).

Reply:
Did you mean res_id is different than res_model_id?
I see in the code k you fetch MODEL from  ID and set ID.. Whats the difference?
@AnuKPatel

Reply:
@Freni-SerpentCS  res_model_id is as 'database id' of the model, Like school_evaluation_template is stored in ir_model table and it's record have id.

res_id is an id of the school_evaluation_template. For Ex. Rating is belongs to 4rth no. evaluation template.

so you need to improve the code here.

Reply:
Was there a code done afterwards?

Reply:
yes, Improvements are done.